lib/internal/http.js

```diff
@@ -7,6 +7,7 @@ const {
 const { setUnrefTimeout } = require('internal/timers');
 const { PerformanceEntry, notify } = internalBinding('performance');
 
+const VALID_PATH_REGEX = /^[_0-9A-Za-z]+(?:\.[_0-9A-Za-z]+)*\.?:(?:[1-9]|[1-9][0-9]{1,2}|[1-5][0-9]{3}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$/;
```

Comment:
Benchmarks tests may be required to assess whether performance has been affected.

Reply:
Could you guide me more about these benchmark tests, as how to write one?

Reply:
@PreYunk Usually there is no need to write additional `benchmarks` code, you can use existing `benchmarks` for testing, refer to the document https://github.com/nodejs/node/blob/master/doc/guides/writing-and-running-benchmarks.md

Reply:
The regex here does not properly account for the possibility of IPv6 addresses. For instance, `[2001:db8::1]:111` should be a valid path but fails this check. A likely better approach for the validation check would be to attempt to create a URL from the path:

```js
new URL(`http://${path}`)
```

If creating the URL is successful, check that the `hostname` and `port` are appropriately set, then continue.

Reply:
Also you need to check that `http://${url.host}` is `url.href` to disallow `asdf.com:1234/asdf` etc.